modules/dnn/src/dnn.cpp

```diff
@@ -191,6 +191,101 @@ void blobFromImages(InputArrayOfArrays images_, OutputArray blob_, double scalef
     }
 }
 
+std::vector<Mat> imagesFromBlob(const cv::Mat& blob_, const bool normalize_,
+                                const cv::Size outputSize_, const double scaleFactor_,
+                                const cv::Scalar& mean_)
+{
+    CV_TRACE_FUNCTION();
+
+    //A blob is a 4 dimensional matrix in floating point precision
+    CV_Assert(blob_.depth() == CV_32F);
+    CV_Assert(blob_.dims == 4);
+
+    //A container to store our images
+    std::vector<cv::Mat> vectorOfImages;
+
+    //Store each dimension size
+    //It is ok to hardcode it since we asserted the dimension before
+    int nbOfImages = blob_.size[0]; //= nb of inputs
+    int nbOfChannels = blob_.size[1]; //<=> nb of filtered images
+    int height = blob_.size[2]; //= the height of the images
+    int width = blob_.size[3]; //= the width of the images
+
+
+    //Access the elements of each channel for each images
+    //Store the matrix in a vector of Images
+    for (int i = 0; i < nbOfImages; i++)
+    {
+        //store all the channels for each image
+        std::vector<cv::Mat> vectorOfChannels;
+
+        for (int c = 0; c < nbOfChannels; c++)
+        {
+            //store the current channel
+            cv::Mat currentChannel(width, height, CV_32F);
+
+            for (int w = 0; w < width; w++)
+            {
+                for (int h = 0; h < height; h++)
+                {
+                    int indx[4] = { i, c, h, w };
+                    currentChannel.at<float>(h, w) = blob_.at<float>(indx); //blobs store images in floating point precision
```

Comment:
That could be done much more easier like
```cpp
for (int i = 0; i < nbOfImages; ++i)
{
    std::vector<Mat> vectorOfChannels(nbOfChannels);
    for (int c = 0; c < nbOfChannels; ++c)
        vectorOfChannels[c] = slice(blob_, n, c);  // function from shape_utils.hpp
}
```

Reply:
Am I right if I assume you wanted to use `Mat getPlane(const Mat &m, int n, int cn)` instead of the quoted `Mat slice(const Mat &m, const _Range &r0, const _Range &r1)`?

Reply:
`slice` can manage it. There is a constructor of cv::Range for a single value, https://docs.opencv.org/master/d9/ddc/structcv_1_1dnn_1_1__Range.html#a5339946e5ddf17692f62956c28cbe623.

Reply:
I agree, but as far as I understand it, `slice` will output a 4D array (1,1,size,size) for each channel. This will result in a (1,c,size,size) array for each image using `cv::merge()`, where c is the number of channels. This is fine if we want to output a blob per image, but it does not allow us to directly use the images with the OpenCV functions made for 2D arrays (e.g. `cv::imshow()` and `cv::imwrite()`).

I propose to add a boolean flag to let the user choose if he prefers to get a 4D or a 2D array per image contained in a blob.

```c++
//slice channels
std::vector<Mat> vectorOfChannels(nbOfChannels);
for (int c = 0; c < nbOfChannels; ++c)
{
	if(get2DArrays_) vectorOfChannels[c] = getPlane(blob_, i, c); // function from shape_utils.hpp in dnn module, a plane is 2D array
	else vectorOfChannels[c] = slice(blob_, i, c); // function from shape_utils.hpp in dnn module, a slice is a 4D array
}//loop on channels
```

Reply:
@Pandinosaurus, Thanks, I see. You're completely right. We need to use `getPlane`:
```cpp
std::vector<int> blobSize(4);
blobSize[0] = 2;  // batch size
blobSize[1] = 3;  // channels
blobSize[2] = 4;  // height
blobSize[3] = 5;  // width
Mat blob(blobSize, CV_32F);

std::vector<Mat> imgs(blob.size[0]);
for (int n = 0; n < blob.size[0]; ++n)
{
    std::vector<Mat> channels(blob.size[1]);
    for (int c = 0; c < blob.size[1]; ++c)
    {
        channels[c] = getPlane(blob, n, c);
    }
    cv::merge(channels, imgs[n]);
}
```